src/algebra/lie/engel.lean

```diff
@@ -0,0 +1,280 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import algebra.lie.nilpotent
+import algebra.lie.cartan_subalgebra
+
+/-!
+# Engel's theorem
+
+This file contains a proof of Engel's theorem providing necessary and sufficient conditions for Lie
+algebras and Lie modules to be nilpotent.
```

Comment:
Please include a more detailed statement here, for people who don't know Engel's theorem yet.

Reply:
Done!